fe/fe-core/src/main/java/com/starrocks/authentication/LDAPAuthenticationProvider.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.starrocks.authentication;
+
+import com.starrocks.analysis.UserIdentity;
+import com.starrocks.mysql.privilege.Password;
+import com.starrocks.mysql.security.LdapSecurity;
+import org.apache.parquet.Strings;
+
+import static com.starrocks.mysql.privilege.AuthPlugin.AUTHENTICATION_LDAP_SIMPLE;
+
+public class LDAPAuthenticationProvider implements AuthenticationProvider {
+    public static final String PLUGIN_NAME = AUTHENTICATION_LDAP_SIMPLE.name();
+
+    @Override
+    public UserAuthenticationInfo validAuthenticationInfo(UserIdentity userIdentity, String password, String textForAuthPlugin)
+            throws AuthenticationException {
+        UserAuthenticationInfo info = new UserAuthenticationInfo();
+        info.setPassword(null);
+        info.setTextForAuthPlugin(textForAuthPlugin);
+        return info;
+    }
+
+    @Override
+    public void authenticate(String user, String host, byte[] password, byte[] randomString,
+                             UserAuthenticationInfo authenticationInfo) throws AuthenticationException {
+        String userForAuthPlugin = authenticationInfo.getTextForAuthPlugin();
+        if (!Strings.isNullOrEmpty(userForAuthPlugin)) {
+            LdapSecurity.checkPassword(userForAuthPlugin, new String(password));
+        } else {
+            LdapSecurity.checkPasswordByRoot(user, new String(password));
+        }
+    }
+
+    @Override
+    public UserAuthenticationInfo upgradedFromPassword(UserIdentity userIdentity, Password password)
```

Comment:
Have we tested upgrade ldap authentication from old version to new version(isUsingNewPrivilege = true) manually?